### PR TITLE
Support gitblit #26

### PR DESCRIPTION
**Pull request by markiewb**
- uses pattern configuration code from my NetBeans plugin https://github.com/markiewb/nb-git-open-in-external-repoviewer

### CODE REVIEW

**Comment by markiewb** [comment]
@ben-gibson: Please review

**Comment by ben-gibson** [comment]
Conflicts with your previous commit.

File: src/uk/co/ben_gibson/repositorymapper/Host/Url/Factory/GitBlit.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+         gitblit.target.show_commitdiff.url.0=<protocol>://<server>/git/commitdiff/<repo|escapeSlashWithBang>.git/<revision>
+         gitblit.target.show_commitdiff.url.1=<protocol>://<server>/commitdiff/<repo|escapeSlashWithBang>.git/<revision>
+         gitblit.target.show_file.url.0=<protocol>://<server>/git/blob/<repo|escapeSlashWithBang>.git/<revision>/<fullfilepath|escapeSlashWithBang>#L<linenumber|1based>
+         gitblit.target.show_file.url.1=<protocol>://<server>/blob/<repo|escapeSlashWithBang>.git/<revision>/<fullfilepath|escapeSlashWithBang>#L<linenumber|1based>
+         * </pre>
+         */
+
+
+        //Note: Use LinkedHashMap because ordering the patterns have a priority
+        Map<String, String> commitDiffPatterns = new LinkedHashMap<>();
```
Comment (ben-gibson) [inline src/uk/co/ben_gibson/repositorymapper/Host/Url/Factory/GitBlit.java]: Would it be more succinct to use an approach similar to the other factories over this patten mapping e.g

```
String path;
String fragment = null;

if (context.getCommitHash() != null) {
    path = String.format("%s/commitdiff/%s", remoteUrl.getPath(), context.getCommitHash());
} else {
    path = String.format(
        "%s/blob/%s%s",
        remoteUrl.getPath(),
        context.getBranch(),
        context.getFilePathRelativeToRepository()
    );

    if (context.getCaretLinePosition() != null) {
        fragment = "L" + context.getCaretLinePosition().toString();
    }
}
```

File: tests/uk/co/ben_gibson/repositorymapper/RepositoryProvider/Host/Url/Factory/GitBlitTest.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+                                32),
+                        "http://my.company.com/gitblit/blob/category!my.complex.repo.git/1234/src!com!foo!FooTest.java#L32"
+                },
+                {
+                        ContextTestUtil.getMockedContext(
+                                "http://git.delwink.com/git/r/ljpapi.git",
+                                "master",
+                                "src/com/delwink/ljp/Alignment.java",
+                                "",
+                                "607a417df243451330d91b48515b4df9eb106580",
```
Comment (ben-gibson) [inline tests/uk/co/ben_gibson/repositorymapper/RepositoryProvider/Host/Url/Factory/GitBlitTest.java]: Are we missing a test for a context with no commit hash, which would produce a URL with the branch?

**Comment by markiewb** [comment]
@ben-gibson:  Please review again! I updated most of the requested parts and I force-pushed the PR.

File: tests/uk/co/ben_gibson/repositorymapper/RepositoryProvider/Host/Url/Factory/GitBlitTest.java (hunk 2 above)
Comment (markiewb) [inline tests/uk/co/ben_gibson/repositorymapper/RepositoryProvider/Host/Url/Factory/GitBlitTest.java]: Test added

File: CHANGELOG.md [hunk 3]
```diff
@@ -15,6 +15,12 @@ All Notable changes to `Open in Git host` will be documented in this file
 
 - Ability to open a specific commit from the VCS log tool window.
 
+<<<<<<< HEAD
```
Comment (ben-gibson) [inline CHANGELOG.md]: Conflicts

File: resources/META-INF/plugin.xml [hunk 4]
```diff
@@ -1,7 +1,11 @@
 <idea-plugin version="2">
   <id>uk.co.ben-gibson.remote.repository.mapper</id>
   <name>Open in Git host</name>
+<<<<<<< HEAD
```
Comment (ben-gibson) [inline resources/META-INF/plugin.xml]: Conflicts.

**Comment by markiewb** [comment]
I will take of it this weekend

Den 9 dec. 2016 13:39 skrev "Ben Gibson" <notifications@github.com>:

**Comment by markiewb** [comment]
@ben-gibson: I removed the artefacts from the previous merge. The CHANGELOG and plugin.xml is updated now.

**Comment by ben-gibson** [comment]
This has now been merged through the command line. I made some minor tweaks to the factory.

**Pull request closed without merging**